### ISSUE TITLE
DIY VIRTUAL DOM IN 20 MINUTES!

### ISSUE THREAD

**Issue by yoshuawuyts** [nodeconf/budapest#73]
#### Title

DIY VIRTUAL DOM IN 20 MINUTES!
#### Description

Ever wondered what goes on inside today's framework engines? In this talk
we'll got over how to implement a virtual DOM, from scratch. We'll start off
with some of the basics around why DOM virtualization is useful; gently moving
up to how to create nodes, how to implement diffing, array reordering and
eventually dive into some of the less understood optimizations such as branch
caching. All packaged up neatly into a 20 minute talk.
#### Learning objectives
- get an idea of why virtual DOMs are used
- understand how node diffing works
- understand which optimizations can be applied after basic diffing
- basically we're going to demistify this whole vdom thing once and for all
  because it's definitely time we did. Everyone uses these things and so few
  people seem to understand how it works and that's not a good thing
#### City of residence

Berlin

**Comment by gergelyke** [comment]
Hello  Yoshua,

We really, really thank you for your participation in the Call for Proposal for NodeConf Budapest 2017!

We're sorry to tell you, that although your topic was really great, it didn't get accepted to our event this year. It wasn't an easy decision, and it took a while for us, but with all the factors that influenced us during the selection process, it had to be made.

To compensate this sad news a bit, we would like to invite you to come as an attendee! As an appreciation for the effort you already made to make this event great, we'd like to give you a ticket with a 20% discount: 
https://ti.to/risingstack/nodeconf-budapest/discount/thanks-for-your-cfp

We hope you can make it!
